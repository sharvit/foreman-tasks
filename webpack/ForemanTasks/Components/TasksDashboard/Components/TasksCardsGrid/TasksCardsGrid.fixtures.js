export const MOCKED_DATA = {
  running: {
    last: 3,
    older: 5,
  },
  paused: {
    last: 3,
    older: 5,
  },
  stopped: {
    error: {
      total: 8,
      last: 1,
    },
    warning: {
      total: 20,
      last: 2,
    },
    success: {
      total: 25,
      last: 3,
    },
  },
  scheduled: 1,
};
