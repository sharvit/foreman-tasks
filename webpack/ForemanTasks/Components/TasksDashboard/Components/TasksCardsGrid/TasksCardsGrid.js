import React from 'react';
import PropTypes from 'prop-types';
import { CardGrid } from 'patternfly-react';
import { noop } from 'foremanReact/common/helpers';

import RunningTasksCard from './Components/RunningTasksCard/RunningTasksCard';
import PausedTasksCard from './Components/PausedTasksCard/PausedTasksCard';
import StoppedTasksCard from './Components/StoppedTasksCard/StoppedTasksCard';
import ScheduledTasksCard from './Components/ScheduledTasksCard/ScheduledTasksCard';

import {
  TASKS_DASHBOARD_AVAILABLE_TIMES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_STATES,
} from '../../TasksDashboardConstants';
import { timePropType, queryPropType } from '../../TasksDashboardPropTypes';

const TasksCardsGrid = ({ time, query, data, updateQuery }) => (
  <CardGrid matchHeight fluid>
    <CardGrid.Row>
      {[
        [TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.RUNNING, RunningTasksCard],
        [TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.PAUSED, PausedTasksCard],
        [TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.STOPPED, StoppedTasksCard],
        [TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.SCHEDULED, ScheduledTasksCard],
      ].map(([key, Card]) => (
        <CardGrid.Col sm={6} lg={3} key={key}>
          <Card
            matchHeight
            data={data[key]}
            query={query}
            time={time}
            updateQuery={updateQuery}
          />
        </CardGrid.Col>
      ))}
    </CardGrid.Row>
  </CardGrid>
);

TasksCardsGrid.propTypes = {
  time: timePropType,
  query: queryPropType,
  data: PropTypes.shape({
    [TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.RUNNING]:
      RunningTasksCard.propTypes.data,
    [TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.PAUSED]:
      PausedTasksCard.propTypes.data,
    [TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.STOPPED]:
      StoppedTasksCard.propTypes.data,
    [TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.SCHEDULED]:
      ScheduledTasksCard.propTypes.data,
  }),
  updateQuery: PropTypes.func,
};

TasksCardsGrid.defaultProps = {
  time: TASKS_DASHBOARD_AVAILABLE_TIMES.H24,
  query: {},
  data: {
    [TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.RUNNING]:
      RunningTasksCard.defaultProps.data,
    [TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.PAUSED]:
      PausedTasksCard.defaultProps.data,
    [TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.STOPPED]:
      StoppedTasksCard.defaultProps.data,
    [TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.SCHEDULED]:
      ScheduledTasksCard.defaultProps.data,
  },
  updateQuery: noop,
};

export default TasksCardsGrid;
