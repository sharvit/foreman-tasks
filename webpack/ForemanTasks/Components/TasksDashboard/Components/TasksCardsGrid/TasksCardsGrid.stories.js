import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, object, select } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import {
  TASKS_DASHBOARD_AVAILABLE_TIMES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_STATES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_RESULTS,
  TASKS_DASHBOARD_AVAILABLE_QUERY_MODES,
} from '../../TasksDashboardConstants';
import { MOCKED_DATA } from './TasksCardsGrid.fixtures';
import TasksCardsGrid from './TasksCardsGrid';

const createSelect = (name, options) =>
  select(name, { ...options, NONE: 'none' }, 'none');

storiesOf('TasksDashboard/TasksCardsGrid', module)
  .addDecorator(withKnobs)
  .add('TasksCardsGrid', () => {
    const selectTime = select(
      'time',
      TASKS_DASHBOARD_AVAILABLE_TIMES,
      TasksCardsGrid.defaultProps.time
    );
    const selectState = createSelect(
      'query.state',
      TASKS_DASHBOARD_AVAILABLE_QUERY_STATES
    );
    const selectResult = createSelect(
      'query.result',
      TASKS_DASHBOARD_AVAILABLE_QUERY_RESULTS
    );
    const selectMode = createSelect(
      'query.mode',
      TASKS_DASHBOARD_AVAILABLE_QUERY_MODES
    );

    return (
      <TasksCardsGrid
        time={selectTime}
        query={{
          state: selectState,
          result: selectResult,
          mode: selectMode,
          time: selectTime,
        }}
        data={object('data', MOCKED_DATA)}
        updateQuery={action('updateQuery')}
      />
    );
  });
