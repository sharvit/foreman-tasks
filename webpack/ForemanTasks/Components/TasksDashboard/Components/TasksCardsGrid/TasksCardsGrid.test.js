import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import {
  TASKS_DASHBOARD_AVAILABLE_TIMES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_STATES,
} from '../../TasksDashboardConstants';
import { MOCKED_DATA } from './TasksCardsGrid.fixtures';
import TasksCardsGrid from './TasksCardsGrid';

const fixtures = {
  'render with minimal props': {},
  'render with props': {
    time: TASKS_DASHBOARD_AVAILABLE_TIMES.WEEK,
    query: { state: TASKS_DASHBOARD_AVAILABLE_QUERY_STATES.RUNNING },
    data: MOCKED_DATA,
    updateQuery: jest.fn(),
  },
};

describe('TasksCardsGrid', () =>
  testComponentSnapshotsWithFixtures(TasksCardsGrid, fixtures));
