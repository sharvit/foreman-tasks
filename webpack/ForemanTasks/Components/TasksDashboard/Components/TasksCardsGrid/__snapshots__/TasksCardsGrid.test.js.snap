// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksCardsGrid render with minimal props 1`] = `
<CardGrid
  className=""
  fluid={true}
  matchHeight={true}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      key="running"
      lg={3}
      sm={6}
    >
      <RunningTasksCard
        className=""
        data={
          Object {
            "last": 0,
            "older": 0,
          }
        }
        matchHeight={true}
        query={Object {}}
        time="H24"
        updateQuery={[Function]}
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      key="paused"
      lg={3}
      sm={6}
    >
      <PausedTasksCard
        className=""
        data={
          Object {
            "last": 0,
            "older": 0,
          }
        }
        matchHeight={true}
        query={Object {}}
        time="H24"
        updateQuery={[Function]}
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      key="stopped"
      lg={3}
      sm={6}
    >
      <StoppedTasksCard
        className=""
        data={
          Object {
            "error": Object {
              "last": 0,
              "total": 0,
            },
            "success": Object {
              "last": 0,
              "total": 0,
            },
            "warning": Object {
              "last": 0,
              "total": 0,
            },
          }
        }
        matchHeight={true}
        query={Object {}}
        time="H24"
        updateQuery={[Function]}
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      key="scheduled"
      lg={3}
      sm={6}
    >
      <ScheduledTasksCard
        className=""
        data={0}
        matchHeight={true}
        query={Object {}}
        time="H24"
        updateQuery={[Function]}
      />
    </Col>
  </Row>
</CardGrid>
`;

exports[`TasksCardsGrid render with props 1`] = `
<CardGrid
  className=""
  fluid={true}
  matchHeight={true}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      key="running"
      lg={3}
      sm={6}
    >
      <RunningTasksCard
        className=""
        data={
          Object {
            "last": 3,
            "older": 5,
          }
        }
        matchHeight={true}
        query={
          Object {
            "state": "running",
          }
        }
        time="WEEK"
        updateQuery={[MockFunction]}
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      key="paused"
      lg={3}
      sm={6}
    >
      <PausedTasksCard
        className=""
        data={
          Object {
            "last": 3,
            "older": 5,
          }
        }
        matchHeight={true}
        query={
          Object {
            "state": "running",
          }
        }
        time="WEEK"
        updateQuery={[MockFunction]}
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      key="stopped"
      lg={3}
      sm={6}
    >
      <StoppedTasksCard
        className=""
        data={
          Object {
            "error": Object {
              "last": 1,
              "total": 8,
            },
            "success": Object {
              "last": 3,
              "total": 25,
            },
            "warning": Object {
              "last": 2,
              "total": 20,
            },
          }
        }
        matchHeight={true}
        query={
          Object {
            "state": "running",
          }
        }
        time="WEEK"
        updateQuery={[MockFunction]}
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      key="scheduled"
      lg={3}
      sm={6}
    >
      <ScheduledTasksCard
        className=""
        data={1}
        matchHeight={true}
        query={
          Object {
            "state": "running",
          }
        }
        time="WEEK"
        updateQuery={[MockFunction]}
      />
    </Col>
  </Row>
</CardGrid>
`;
